File: VB/CustomEditorSample/PictureBoxSizingEditorViewModel.vb
Imports Microsoft.VisualBasic
Imports System
Imports System.Collections.Generic
Imports System.ComponentModel
Imports System.Linq.Expressions
Imports System.Windows.Input
Imports DevExpress.Utils
Imports DevExpress.Xpf.Core.Commands
Imports DevExpress.Xpf.Core
Imports DevExpress.Xpf.ReportDesigner
Imports DevExpress.XtraPrinting

Namespace CustomEditorSample
	Public Class PictureBoxSizingEditorViewModel
		Implements INotifyPropertyChanged
		Private ReadOnly editorServices As EditorServices
		Private sizeMode As ImageSizeMode
		Private ReadOnly applyCommand_Renamed As ICommand
		Private ReadOnly controlName As String

		Public Shared ReadOnly Property ImageSizeModes() As Array
			Get
				Return EnumExtensions.GetValues(GetType(ImageSizeMode))
			End Get
		End Property

		Public Property ImageSizeMode() As ImageSizeMode
			Get
				Return sizeMode
			End Get
			Set(ByVal value As ImageSizeMode)
                If Not sizeMode = value Then
                    sizeMode = value
                    RaisePropertyChanged(Function() sizeMode)
                End If
			End Set
		End Property

		Public ReadOnly Property ApplyCommand() As ICommand
			Get
				Return applyCommand_Renamed
			End Get
		End Property

		Public Sub New(ByVal editorServices As EditorServices, ByVal controlInfo As ReportControlInfo, ByVal customProperties As IDictionary(Of String, Object))
			Me.editorServices = editorServices
			applyCommand_Renamed = New DelegateCommand(Of Object)(AddressOf Apply)
			controlName = controlInfo.DesignName
			Const customProperyName As String = "Sizing"
			ImageSizeMode = If(customProperties.ContainsKey(customProperyName) AndAlso TypeOf customProperties(customProperyName) Is ImageSizeMode, CType(customProperties(customProperyName), ImageSizeMode), DevExpress.XtraPrinting.ImageSizeMode.Normal)
		End Sub

		Private Sub Apply(ByVal parameter As Object)
            Dim changeService As DevExpress.Xpf.ReportDesigner.Services.IChangeService = editorServices.ChangeService
			changeService.TriggerCustomAction(New ChangeImageSizeModeAction() With {.ControlName = controlName, .ImageSizeMode = ImageSizeMode})
			changeService.CommitChanges()
		End Sub
		#Region "INotifyPropertyChanged"
		Public Event PropertyChanged As PropertyChangedEventHandler Implements INotifyPropertyChanged.PropertyChanged
		Private Sub RaisePropertyChanged(Of T)(ByVal [property] As Expression(Of Func(Of T)))
			PropertyExtensions.RaisePropertyChanged(Me, PropertyChangedEvent, [property])
		End Sub
		#End Region
	End Class
End Namespace

File: CS/CustomEditorSample/PictureBoxSizingEditorViewModel.cs
using System;
using System.ComponentModel;
using DevExpress.Xpf.ReportDesigner;
using DevExpress.XtraPrinting;
using DevExpress.Xpf.Core;
using System.Linq.Expressions;
using DevExpress.Utils;
using System.Windows.Input;
using DevExpress.Xpf.Core.Commands;
using System.Collections.Generic;

namespace CustomEditorSample {
    public class PictureBoxSizingEditorViewModel : INotifyPropertyChanged {
        readonly EditorServices editorServices;
        ImageSizeMode sizeMode;
        readonly ICommand applyCommand;
        readonly string controlName;

        public static Array ImageSizeModes { get { return EnumExtensions.GetValues(typeof(ImageSizeMode)); } }

        public ImageSizeMode ImageSizeMode {
            get { return sizeMode; }
            set {
                if(sizeMode != value) {
                    sizeMode = value;
                    RaisePropertyChanged(() => sizeMode);
                }
            }
        }

        public ICommand ApplyCommand { get { return applyCommand; } }

        public PictureBoxSizingEditorViewModel(EditorServices editorServices, ReportControlInfo controlInfo, IDictionary<string, object> customProperties) {
            this.editorServices = editorServices;
            applyCommand = new DelegateCommand<object>(Apply);
            controlName = controlInfo.DesignName;
            const string customProperyName = "Sizing";
            ImageSizeMode = customProperties.ContainsKey(customProperyName) && customProperties[customProperyName] is ImageSizeMode ?
                (ImageSizeMode)customProperties[customProperyName] : DevExpress.XtraPrinting.ImageSizeMode.Normal;
        }

        void Apply(object parameter) {
            var changeService = editorServices.ChangeService;
            changeService.TriggerCustomAction(new ChangeImageSizeModeAction() { ControlName = controlName, ImageSizeMode = ImageSizeMode });
            changeService.CommitChanges();
        }
        #region INotifyPropertyChanged
        public event PropertyChangedEventHandler PropertyChanged;
        void RaisePropertyChanged<T>(Expression<Func<T>> property) {
            PropertyExtensions.RaisePropertyChanged(this, PropertyChanged, property);
        }
        #endregion
    }
}
